libs/runtime/ebpf_hash_table.c

```diff
@@ -189,6 +189,21 @@ _ebpf_hash_table_compare(_In_ const ebpf_hash_table_t* hash_table, _In_ const ui
     if (length_a_in_bits > length_b_in_bits) {
         return 1;
     }
+
+    if (!hash_table->extract) {
+        // Special case common key sizes.
+        switch (length_a_in_bits) {
+        case 8:
+            return (uint8_t)*data_a - (uint8_t)*data_b;
+        case 16:
+            return (uint16_t) * (uint16_t*)data_a - (uint16_t) * (uint16_t*)data_b;
+        case 32:
+            return (uint32_t) * (uint32_t*)data_a - (uint32_t) * (uint32_t*)data_b;
+        case 64:
+            return (uint32_t)((uint64_t) * (uint64_t*)data_a - (uint64_t) * (uint64_t*)data_b);
```

Comment:
since we are truncating to 32 bits, it can incorrectly return 0 if lower 32 bits of the difference are 0?

Reply:
Good point.